### ISSUE TITLE
attention maps

### ISSUE THREAD

**Issue by soldierofhell** [fizyr/keras-retinanet#959]
Hi, has anyone tried to apply attention maps to classification output? Preferable with https://github.com/raghakot/keras-vis. I wonder if in general it is applicable to object detection, because all examples of such visualization are for classification task

**Comment by freaad** [comment]
I am also interested in this one. I face so many false positve about my custom network.

**Comment by martinzlocha** [comment]
I have added attention maps to gate the features from the skip connections based on this paper:

https://arxiv.org/abs/1804.03999

It improves the accuracy but don't expect too much, maybe 1% mAP improvement.

Here is the code for the Attention block:

```
def AttentionBlock(x, g):
    g1 = keras.layers.BatchNormalization()(g)
    x1 = keras.layers.BatchNormalization()(x)

    g1_x1 = keras.layers.Add()([g1, x1])
    psi = keras.layers.Activation('relu')(g1_x1)
    psi = keras.layers.Conv2D(1, kernel_size=1)(psi)
    psi = keras.layers.BatchNormalization()(psi)
    psi = keras.layers.Activation('sigmoid')(psi)
    x = keras.layers.Multiply()([x, psi])
    return x
```

In this case x is the output of the skip connection and g (gating signal) is the upsampled feature map. 

I also tried to gate the features that get passed into the subnetworks but that didn't help.

**Comment by hgaiser** [comment]
Since there is no issue with our code here, I'll close this issue. Feel free to continue discussing on the Slack channel.
